### DISCUSSION TITLE
cargo run `Failed to fetch tree-sitter grammars`

### DISCUSSION THREAD

**Discussion by bogdan-ov** [helix-editor/helix#9650] [Q&A] [answered]
Hi! I want to play with the helix code, but when I run `cargo run` (also `build` and `build --release`) command in the root dir, I get this output:
```
   Compiling helix-term v23.10.0 (***/helix/helix-term)
error: failed to run custom build command for `helix-term v23.10.0 (***/helix/helix-term)`

Caused by:
  process didn't exit successfully: `***/helix/target/debug/build/helix-terme6ff343e52fec56a/build-script-build` (exit status: 101)
  --- stderr
  thread 'main' panicked at 'Failed to fetch tree-sitter grammars: data did not match any variant of untagged enum GrammarSource
  in `grammar.source`
  ', helix-term/build.rs:5:26
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
How can I fix this?

What I have:
- arch linux 6.7.4
- cargo 1.70.0
- also I have `gcc` compiler (as I know treesitter needs some kind of C compiler)

I'm sorry if this is a stupid question, but I really have no idea how to fix this...
**Thank you in advance!**

**Comment by the-mikedavis** [comment]
Do you have any custom config in `~/.config/helix/languages.toml`, `.helix/languages.toml` or modifications to the `languages.toml` in the root of the repository?

**Reply by bogdan-ov** [reply]
Yes, I have only custom `~/.config/helix/languages.toml`, I haven't modified the source code in any way
Should I delete my `languages.toml` from config dir? (or just move/rename it)

**Reply by the-mikedavis** [reply, marked as answer]
Yeah, the error message is saying that there's a `[[grammar]]` entry with an invalid `source` field in the language config. If you move or delete that file the build should work

**Reply by bogdan-ov** [reply]
Yea, this helped! Thank you very much!
